lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -265,6 +293,54 @@ defmodule BroadwayRabbitMQ.AmqpClient do
       {:error, :timeout}
   end
 
+  defp get_channel(%{connection: nil} = config) do
+    case config[:custom_pool][:checkout] do
+      {m, f, a} -> apply(m, f, [self() | a])
+      fun when is_function(fun, 1) -> fun.(self())
+      _ -> {:error, ":checkout must be a mfa or a 1-arity function"}
+    end
```

Comment:
I don't think we need to pass `self()` to the custom pool's `checkout` function, since that function can call `self()` itself if it wants to, right?